include/xtensor/xfunction.hpp

```diff
@@ -163,8 +163,11 @@ namespace xt
         using stepper = typename iterable_base::stepper;
         using const_stepper = typename iterable_base::const_stepper;
 
-        static constexpr layout_type static_layout = compute_layout(std::decay_t<CT>::static_layout...);
-        static constexpr bool contiguous_layout = detail::conjunction_c<std::decay_t<CT>::contiguous_layout...>::value;
+        static constexpr layout_type static_layout = sizeof...(CT) != 0 ? compute_layout(std::decay_t<CT>::static_layout...) : layout_type::any;
+        static constexpr bool contiguous_layout = sizeof...(CT) != 0 && detail::conjunction_c<std::decay_t<CT>::contiguous_layout...>::value;
+        static constexpr bool no_broadcast = sizeof...(CT) != 0 &&
+                                             contiguous_layout &&
+                                             detail::equal_dimensions<typename std::decay_t<CT>::shape_type...>::value;
```

Comment:
These definitions are becoming complicated enough so we can encapsulate them in dedicated metafunctions.

Reply:
how would you do that? add the values as arguments to another template?

Reply:
I think a traits class for these three properties would limit the number of template instantiated:
```cpp
templte <class... CT>
struct xfunciton_layout
{
    static constexpr layout_type static_layout = ...;
    static constexpr bool contiguous_layout = ....;
    static constexpr bool no_broadcast = ....;
};
```